File: design-patterns/strategy/html_formatter.rb
class HTMLFormatter
  def output_report(context)
    p '<html>'
    p '<head>'
    p "<title>#{context.title}</title>"
    p '</head>'
    p '<body>'
    context.text.each do |line|
      p "<p>#{line}</p>"
    end
    p '</body>'
    p '</html>'
  end
end

File: design-patterns/cs/Patrones/Patrones/Strategy/HTMLFormatter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Patrones.Strategy
{
    public class HTMLFormatter : IFormatStrategy
    {
        public string OutputReport(Report context)
        {
            string aux = string.Empty;

            aux += "<html>";
            aux += "<head>";
            aux += "<title>";
            aux += context.Title;
            aux += "</title>";
            aux += "</head>";

            aux += "<body>";
            foreach (var item in context.Body)
            {
                aux += "<p>" + item + "</p>";
            }
            aux += "</body>";

            aux += "</html>";
            return aux;
        }
    }
}
